launcher/src/main/java/org/apache/spark/launcher/LauncherServer.java

```diff
@@ -320,7 +320,7 @@ protected void handle(Message msg) throws IOException {
         } else {
           if (handle == null) {
             throw new IllegalArgumentException("Expected hello, got: " +
-              msg != null ? msg.getClass().getName() : null);
+              (msg != null ? msg.getClass().getName() : "null"));
```

Comment:
You could probably just improve this by creating this class name string once above line 321, and putting something like "no message" instead of null, while we're here.

Reply:
Sure